src/Components/OATGraphViewer/OATGraphViewer.tsx

```diff
@@ -637,15 +639,45 @@ const OATGraphViewer = ({ state, dispatch }: OATGraphProps) => {
         setCurrentLocation(flowTransform);
     }, []);
 
+    const getNewNodePosition = (coordinates) => {
+        // Find the amount of nodes at the same position
+        const nodesAtPosition = elements.filter(
+            (element) =>
+                !element.source &&
+                element.position.x === coordinates.x &&
+                element.position.y === coordinates.y
+        );
+
+        // If there is no node at the same position, return the coordinates
+        if (nodesAtPosition.length === 0) {
+            return coordinates;
+        }
+        // Define the new coordinates
+        const newCoordinates = {
+            x: coordinates.x + nodesAtPosition.length * 10,
+            y: coordinates.y + nodesAtPosition.length * 10
+        };
+        // Prevent nodes with the same position
+        return getNewNodePosition(newCoordinates);
+    };
+
     const onNewModelClick = () => {
         if (!state.modified) {
             // Create a new floating node
             const name = `Model${newModelId}`;
             const id = `${idClassBase}model${newModelId};${versionClassBase}`;
+            let startPositionCoordinates = null;
+            if (newModelButtonRef.current) {
```

Comment:
If current is null, then startPositionCoordinates will be null too. What happens if we then call getNewNodePosition with a null value for the param?